process/smartContract/process.go

```diff
@@ -1534,6 +1541,7 @@ func (sc *scProcessor) createSCRsWhenError(
 	}
 
 	accumulatedSCRData := ""
+	// TODO: check why just ESDTTransfer here. No problem with burn, freeze? Can't they be crossShard?
```

Comment:
the return data in case of burn and freeze is handled otherwise. The tokens are sent back through ESDTTransfer in case of error from metachain sc.

Reply:
ok!